### PR TITLE
Hide permission controls for personal collection's sub-collections

### LINKED ISSUE #8406: Unable to share sub-collections from Personal Collection
### Bugs
If you're filing an issue about a bug please include as much information
as you can including the following.

- Your browser and the version: Chrome 68.0.3440.106
- Your operating system: OS X 10
- Your databases: Redshift
- Metabase version: 0.30.1
- Metabase hosting environment: ECS/Docker 
- Metabase internal database: Postgres

## Repro Steps:
1. Create a sub-collection in your Personal Collection
2. Navigate into the sub-collection, hit Edit the permissions for this collection - at this point no one has access to this collection:  
image

3. Make a change granting another group view access: 
image

4. **Issue 1:** Get message about All Users already having greater access? Doesn't make much sense... 
image

5. **Issue 2:** Hit Revoke access, then Save, get this message: 
image

6. Logs show: 
```PUT /api/collection/graph 500 (71 ms) (7 DB calls).
{:message "You cannot edit permissions for a Personal Collection or its descendants.",
:type java.lang.Exception,
:stacktrace
["models.permissions$check_not_personal_collection_or_descendant.invokeStatic(permissions.clj:474)"
"models.permissions$check_not_personal_collection_or_descendant.invoke(permissions.clj:461)"
"models.permissions$revoke_collection_permissions_BANG_.invokeStatic(permissions.clj:479)"
"models.permissions$revoke_collection_permissions_BANG_.invoke(permissions.clj:476)"
"models.collection$fn__27294$update_collection_permissions_BANG___27299$fn__27300.invoke(collection.clj:878)"
"models.collection$fn__27294$update_collection_permissions_BANG___27299.invoke(collection.clj:870)"
"models.collection$fn__27319$update_group_permissions_BANG___27324$fn__27325.invoke(collection.clj:887)"
"models.collection$fn__27319$update_group_permissions_BANG___27324.invoke(collection.clj:884)"
"models.collection$fn__27360$update_graph_BANG___27369$fn__27370$fn__27374.invoke(collection.clj:914)"
"models.collection$fn__27360$update_graph_BANG___27369$fn__27370.invoke(collection.clj:912)"
"models.collection$fn__27360$update_graph_BANG___27369.invoke(collection.clj:902)"
"api.collection$fn__46474.invokeStatic(collection.clj:304)"
"api.collection$fn__46474.invoke(collection.clj:299)"
"middleware$enforce_authentication$fn__55250.invoke(middleware.clj:116)"
"api.routes$fn__55392.invokeStatic(routes.clj:62)"
"api.routes$fn__55392.invoke(routes.clj:62)"
"routes$fn__55477$fn__55478.doInvoke(routes.clj:108)"
"routes$fn__55477.invokeStatic(routes.clj:103)"
"routes$fn__55477.invoke(routes.clj:103)"
"middleware$catch_api_exceptions$fn__55379.invoke(middleware.clj:424)"
"middleware$log_api_call$fn__55356$fn__55358.invoke(middleware.clj:351)"
"middleware$log_api_call$fn__55356.invoke(middleware.clj:350)"
"middleware$add_security_headers$fn__55304.invoke(middleware.clj:253)"
"core$wrap_streamed_json_response$fn__60666.invoke(core.clj:67)"
"middleware$bind_current_user$fn__55255.invoke(middleware.clj:140)"
"middleware$maybe_set_site_url$fn__55308.invoke(middleware.clj:277)"]}
```

I'm really hoping the intention is that we're able to share sub-collections from our Personal Collection with different groups (assuming that's why the option exists in the UI). There's already been a lot of confusion about this after we upgraded this morning, right now everyone is just saving everything in the 'Migrated Questions' collections to get around this.

### PR DESCRIPTION
**Pull request by kulyk**
According to our [docs](https://www.metabase.com/docs/latest/administration-guide/06-collections.html#personal-collections): "A personal collection works just like any other collection except that its permissions can’t be changed". The issue is that we still display a link to the collection permissions modal for personal sub-collections. This PR makes sure the modal won't be shown for personal collections and their sub-collections via UI and via `/permissions` URL

Closes #8406

### To Verify

1. Open your personal collection
2. Ensure there is no link to the collection permission modal (lock icon)
3. Create a collection inside your personal collection
4. Ensure there is no link to the collection permission modal (lock icon)
5. Try opening the modal by adding `/permissions` to an end of a URL. You have to be navigated back to `/collection/:id` page

### Demo

https://user-images.githubusercontent.com/17258145/122540129-fc9bab00-d030-11eb-9fbb-31daa31ae08b.mov

### CODE REVIEW

File: frontend/test/metabase/scenarios/collections/personal-collections.cy.spec.js [hunk 1]
```diff
@@ -52,6 +52,24 @@ describe("personal collections", () => {
...
+
+      // Check can't open permissions modal via URL for personal collection
+      cy.findByText("Your personal collection").click();
+      cy.location().then(location => {
+        cy.visit(`${location}/permissions`);
+        cy.get(".Modal").should("not.exist");
+        cy.url().should("eq", String(location));
+
+        // Check can't open permissions modal via URL for personal collection child
+        cy.get("[class*=CollectionSidebar]")
```
Comment (alxnddr) [inline frontend/test/metabase/scenarios/collections/personal-collections.cy.spec.js:64]: nit: the tree has role `tree`

**Review by alxnddr — APPROVED** [review]
Looks and works good 👍

File: frontend/test/metabase/scenarios/collections/personal-collections.cy.spec.js (hunk 1 above)
Comment (kulyk) [reply frontend/test/metabase/scenarios/collections/personal-collections.cy.spec.js]: Goot point, just copy-pasted the existed selector here. Switched to `sidebar()` helper

**Review by flamber — APPROVED** [review]
This works :+1: 
One note, if you are non-admin, then it doesn't hide the `../permissions` modal of one of their PC sub-collections. It works on their main PC.
It's not blocking this PR, since there's already an issue (15354) for not showing those in general

**Pull request merged**